aloha-core/src/test/scala/com/eharmony/aloha/models/multilabel/MultilabelModelTest.scala

```diff
@@ -47,124 +63,336 @@ class MultilabelModelTest extends ModelSerializationTestHelper {
-    // Subvalue.audited.value should be Some(value2).
-    // 'value' should equal 'value2'.
-    // Check the errors and missing values.
+    val predictions = Map("label" -> 1.0)
+    val report   = reportSuccess(
+      modelId    = ModelId(),
+      labelInfo  = labelsAndInfoEmpty.copy(labelsNotInTrainingSet = labelsNotInTrainingSet),
+      missing    = scm.Map("" -> labelsNotInTrainingSet),
+      prediction = predictions,
+      auditor    = Auditor
+    )
+    assertEquals(Some(predictions), report.natural)
+    assertEquals(Some(predictions), report.audited.value)
+    assertEquals(report.natural, report.audited.value)
+  }
+
+  @Test def testLabelsForPredictionContainsProblemsWhenNoLabelProvided(): Unit = {
+    val labelsAndInfoNoLabels = labelsForPrediction(
+      example          = Map[Label, Label](),  // no label provided
+      labelsOfInterest = labelsOfInterestExtractor,
+      labelToInd       = labelsInTrainingSetToIndex)
 
-    fail()
+    // In this scenario no problems are found but an
+    // Option(GenAggFuncAccessorProblems) is returned instead of None
+    val problemsNoLabelsExpected = Option(GenAggFuncAccessorProblems(Seq(), Seq()))
+    assertEquals(problemsNoLabelsExpected, labelsAndInfoNoLabels.problems)
   }
 
-  @Test def testLabelsForPredictionContainsProblemsWhenLabelsIsEmpty(): Unit = {
-    // Test this:
-    //    val problems =
-    //      if (labelsShouldPredict.nonEmpty) None
-    //      else Option(labelsOfInterest.accessorOutputProblems(a))
+  @Test def testLabelsForPredictionContainsProblemsWhenLabelsIsNotPresent(): Unit = {
+    val labelsExtractor =
+      (label: Map[Label, Label]) => label.get("labels") match {
+        case ls: sci.IndexedSeq[_] => Option(ls.asInstanceOf[sci.IndexedSeq[Label]])
+        case _ => None
+      }
+
+    val descriptor = "label is missing"
+    val labelsOfInterest =
+      GenFunc.f1(GeneratedAccessor(descriptor, labelsExtractor, None))(
+        "", _ getOrElse sci.IndexedSeq.empty[Label]
+      )
+
+    val labelsAndInfoNoLabelsGen1 = labelsForPrediction(
+      example          = Map[Label, Label](),
+      labelsOfInterest = labelsOfInterest,
+      labelToInd       = labelsInTrainingSetToIndex)
+    val problemsNoLabelsExpectedGen1 =
+      Option(GenAggFuncAccessorProblems(Seq(descriptor), Seq()))
+    assertEquals(problemsNoLabelsExpectedGen1, labelsAndInfoNoLabelsGen1.problems)
+  }
 
-    fail()
+  @Test def testLabelsForPredictionContainsProblemsWhenLabelAccessorThrows(): Unit = {
+    val descriptor = "label accessor that throws"
+    val labelsOfInterest =
+      GenFunc.f1(GeneratedAccessor(descriptor,
+        ( _ => throw new Exception()) : Map[String, String] => Option[sci.IndexedSeq[String]], None))(
+        "", _ getOrElse sci.IndexedSeq.empty[Label]
+      )
+    val labelsOfInterestWrapped = EnrichedErrorGenAggFunc(labelsOfInterest)
+
+    val problemsNoLabelsExpected = Option(GenAggFuncAccessorProblems(Seq(), Seq(descriptor)))
+    Try(
```

Comment:
Can you just do something like what follows?  It seems like it would be cleaner.

```scala
Try { 
  labelsForPrediction(Map.empty, labelsOfInterestWrapped, labelsInTrainingSetToIndex)
} match {
  // Failure is scala.util.Failure
  case Failure(SemanticsUdfException(_, _, _, accessorsInErr, _, _)) => 
    assertEquals(accessorsInErr, problemsNoLabelsExpected.get.errors)
  case _ => fail()
}
```